EIPS/eip-beacon_chain_push_withdrawals.md

```diff
@@ -0,0 +1,130 @@
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-02-28
+---
+
+## Abstract
+
+Introduce a new [EIP-2718 transaction type](./eip-2718.md) to support validator withdrawals that are "pushed" from the beacon chain to the EVM.
+
+Add block validations to ensure the withdrawal transactions are sound with respect to withdrawal processing on the beacon chain.
+
+## Motivation
+
+This EIP provides a way for validator withdrawals made on the beacon chain to enter into the EVM.
+The architecture is "push"-based, rather than "pull"-based, where withdrawals are required to be processed in the execution block as soon as they are dequeued from the beacon chain.
+
+This approach is more involved than "pull"-based alternatives (e.g. [EIP-4788](./eip-4788.md) + user-space withdrawal contract) with respect to the core protocol (by providing a new transaction type with special semantics) but does provide tighter integration of a critical feature into the protocol itself.
+
+## Specification
+
+| constants                     | value                                          | units
+|---                            |---                                             |---
+| `FORK_TIMESTAMP`              | TBD                                            |
+| `WITHDRAWAL_TX_TYPE`          | `0x3`                                          | byte
+
+Beginning with the execution timestamp `FORK_TIMESTAMP`, execution clients **MUST** introduce the following extensions to transaction processing and block validation:
+
+### New transaction type
+
+Define a new [EIP-2718](./eip-2718.md) transaction type with `TransactionType` `WITHDRAWAL_TX_TYPE`.
+
+The `TransactionPayload` is an SSZ-encoded container given by the following schema:
+
+```python
+class WithdrawalTransaction(Container):
+    address: ExecutionAddress
+    amount: EncodedWei
+```
+
+where `ExecutionAddress` is an alias for a `Bytes20` SSZ type and `EncodedWei` is an alias for a `Bytes32` SSZ type.
+
+Refer to the [SSZ specs](https://github.com/ethereum/consensus-specs/blob/master/ssz/simple-serialize.md) for further details on layout and encoding.
+
+### Block validity
+
+If a block contains *any* transactions with `WITHDRAWAL_TX_TYPE` type, they **MUST** come before **ALL** other transactions in the block.
+
+If the execution client receives a block where this is not the case, it **MUST** consider the block invalid.
+
+### Transaction processing
+
+When processing a transaction with `WITHDRAWAL_TX_TYPE` type, the implementation should increase the balance of the `address` specified by
+the `WithdrawalTransaction` by the `amount` of wei specified.
+
+This balance change is unconditional and **MUST** not fail.
+
+This transaction type has no associated gas costs.
+
+Implementations should take care to decode the correct amount of wei from the `EncodedWei` value, which represents a 256-bit unsigned integer value in little-endian byte order.
+
+TODO: add logs?
```

Comment:
Does this transaction have a receipt? Current APIs assume receipts can be accessed by tx-index, so those would break if this transaction does not output a receipt. Also, if the receipt is defined as the standard `type ++ rlp([status, cumulative_transaction_gas_used, logs_bloom, logs])`, then the receipt is a constant bytestring, since this always succeeds, has 0 gas, and no logs.

Reply:
i think it is likely we add some kind of logging and along with those changes we'd introduce some kind of receipt format as well.